### PR TITLE
Casey fix leakage

### PR DESCRIPTION
**Pull request by sometimescasey**
@PsiPhiTheta @antoineviscardi 

Hey gents - I've created a new function in `dataset_utils.py` called `load_clean_train_test()` to do what we discussed - i.e. splitting a single sig_id's data up into training = {a subset of genuine}, test = {the remaining unused genuine + all the fakes}. Uses sklearn's train/test splitting function, with a `random state = 4` set by default. It's based pretty heavily on Antoine's old one.

The previous one has been renamed to `old_load_clean_train()`, just in case we need it. Can remove it later if all is good.

There's some tests (`split_test()`) in place to ensure this is working properly, please feel free to suggest additional ones.

`vanilla_vae.py` and `experiments.py` currently both call `x_train, y_train, x_test, y_test = load_clean_train_test()` within their respective `load_data()` fxns to grab their data: 

- `vanilla_vae.py` only uses x_train;
- `experiments.py` only uses x_test and y_test

So we can see this is both wasteful and would technically be incorrect **if** we didn't set a random state. But we do, so this is ok, even if it's a little slower than it needs to be.

I kicked off VAE training last night for all ~60 speakers using your default parameters (128, 512, 256 , 250 epochs etc) - it OOMs alot but but I can just keep restarting.

Feedback away.

### CODE REVIEW

File: cmiyc/dataset_utils.py [hunk 1]
```diff
@@ -27,6 +37,166 @@ def load_clean_train(sig_type='all', sig_id='all', id_as_label='false'):
...
+
+    x_train = np.vstack(vae_df_train['sig'].to_numpy())
+    y_train = vae_df_train[labels].to_numpy().astype('int')
+
+    assert list(set(y_train)) == [TYPE_MAP[vae_sig_type]], "y_train contains bad values: {}".format(list(set(y_train)))
+
+    x_test = np.vstack(exp_df_test['sig'].to_numpy())
+    y_test = exp_df_test[labels].to_numpy().astype('int')
+
+    assert (1 in list(set(y_test)) and 0 in list(set(y_test))), "y_test contains bad values: {}".format(list(set(y_test)))
```
Comment (antoineviscardi) [inline cmiyc/dataset_utils.py:99]: What does this assertion do? It seems like it would trigger when `y_test` contains only 0s or 1s?

Comment (sometimescasey) [reply cmiyc/dataset_utils.py:99]: check that y_test contains both 1 (genuine) and 0 (fakes), otherwise throw AssertionError

Comment (sometimescasey) [reply cmiyc/dataset_utils.py:99]: and yeah...it would be bad to have a test set that only has a certain class, right?

File: cmiyc/vanilla_vae.py [hunk 2]
```diff
@@ -63,11 +68,14 @@ def vae_loss(inputs, outputs, original_dim, z_mean, z_log_var):
         kl_loss = -0.5 * K.sum(kl_loss, axis=-1)
         return K.mean(reconstruction_loss + kl_loss)
 
-    def get_data(self, sig_id=1, sig_type='genuine'):
+    def load_data(self, sig_id=1, sig_type='genuine'):
         '''
         Load the specified sig id and signature type.
+        
+        TODO: again, wasteful to be calling this here.
```
Comment (antoineviscardi) [inline cmiyc/vanilla_vae.py:75]: Maybe we could decouple the `VanillaVae` and the `load_data` functionality simply by having VanillaVae's train function accept `x_train` and `y_train` as parameters? That is nothing critical tho

**Review by antoineviscardi — APPROVED** [review]

File: cmiyc/vanilla_vae.py (hunk 2 above)
Comment (sometimescasey) [reply cmiyc/vanilla_vae.py:75]: ^ agreed, ideally we should have something that wraps around the VanillaVAE training and the experiment. But as it currently stands, since we save the training weights and load them into the experiment later, this works, even if it's inefficient. Totally fine w you refactoring this later (or I can do it if there isn't something more urgent)

File: cmiyc/dataset_utils.py (hunk 1 above)
Comment (antoineviscardi) [reply cmiyc/dataset_utils.py:99]: Oh yeah you're right! We want to make sure the test set has both class :+1:
